dqm-root/src/common/AMC_histogram.cxx

```diff
@@ -23,32 +23,34 @@ class AMC_histogram: public Hardware_histogram
           m_gebsH[i] = 0;
       }
       m_dir->cd();
-      AMCnum     = new TH1F("AMCnum", "AMC number", 12,  0, 12);
+      AMCnum     	 = new TH1F("AMCnum", "AMC number", 12,  0, 12);
       //L1A        = new TH1F("L1A", "L1A ID", 0xffffff,  0x0, 0xffffff);
-      BX         = new TH1F("BX", "BX ID", 4095,  0x0, 0xfff);
+      L1A_mismatch = new TH1F("L1A_mismatch", "L1A_mismatch (AMC L1A - AMC13 LV1_idT)", 40,  -20.0, 20.0);
+      BX           = new TH1F("BX", "BX ID", 4095,  0x0, 0xfff);
       //Dlength    = new TH1F("Dlength", "Data Length", 0xfffff,  0x0, 0xfffff);
-      FV         = new TH1F("FV", "Format Version", 15,  0x0, 0xf);
-      Rtype      = new TH1F("Rtype", "Run Type", 15,  0x0, 0xf);
-      Param1     = new TH1F("Param1", "Run Param 1", 256,  -0.5, 255.5);
-      Param2     = new TH1F("Param2", "Run Param 2", 256,  -0.5, 255.5);
-      Param3     = new TH1F("Param3", "Run Param 3", 256,  -0.5, 255.5);
-      Onum       = new TH1F("Onum", "Orbit Number", 0xffff,  0, 0xffff);
-      BID        = new TH1F("BID", "Board ID", 0xffff,  0, 0xffff);
-      GEMDAV     = new TH1F("GEMDAV", "GEM DAV list", 24,  0, 24);
-      Bstatus    = new TH1F("Bstatus", "Buffer Status", 24,  0, 24);
-      GDcount    = new TH1F("GDcount", "GEM DAV count", 32,  0, 32);
-      Tstate     = new TH1F("Tstate", "TTS state", 15,  0, 15);
-      ChamT      = new TH1F("ChamT", "Chamber Timeout", 24, 0, 24);
-      OOSG       = new TH1F("OOSG", "OOS GLIB", 1, 0, 1);
+      FV         	 = new TH1F("FV", "Format Version", 15,  0x0, 0xf);
+      Rtype      	 = new TH1F("Rtype", "Run Type", 15,  0x0, 0xf);
+      Param1     	 = new TH1F("Param1", "Run Param 1", 256,  -0.5, 255.5);
+      Param2     	 = new TH1F("Param2", "Run Param 2", 256,  -0.5, 255.5);
+      Param3     	 = new TH1F("Param3", "Run Param 3", 256,  -0.5, 255.5);
+      Onum       	 = new TH1F("Onum", "Orbit Number", 0xffff,  0, 0xffff);
+      BID        	 = new TH1F("BID", "Board ID", 0xffff,  0, 0xffff);
+      GEMDAV     	 = new TH1F("GEMDAV", "GEM DAV list", 24,  0, 24);
+      Bstatus    	 = new TH1F("Bstatus", "Buffer Status", 24,  0, 24);
+      GDcount    	 = new TH1F("GDcount", "GEM DAV count", 32,  0, 32);
+      Tstate     	 = new TH1F("Tstate", "TTS state", 15,  0, 15);
+      ChamT      	 = new TH1F("ChamT", "Chamber Timeout", 24, 0, 24);
+      OOSG       	 = new TH1F("OOSG", "OOS GLIB", 1, 0, 1);
     }
 
      //!Fills the histograms for AMC data
     /*!
     This fills the histograms for the following data: AMC number, L1A number, Bunch Crossing ID, Data Length, Format Version, Run Type, Param 1, 2 and 3, Orbit Number, Board ID, GEM DAV list, Buffer Status, GEM DAV count, TTS state, Chamber Timeout, OOS GLIB sync, CRC, L1AT, and DlengthT
     */
-    void fillHistograms(AMCdata *amc){
+    void fillHistograms(AMCdata *amc, AMC13Event *amc13){
```

Comment:
Do you really need to keep the whole amc13 object here? Pass only the value you need.

Reply:
I actually wanted to keep it clean, passing only one object instead of a list of the ones that I'm using... If you prefer it the other way around, I can change it.
Is there a technical specific reason why not to pass the entire object? Just to learn...